src/http/Request.ts

```diff
@@ -1,27 +1,33 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License.
 
-import { HttpMethod, HttpRequest } from '@azure/functions';
+import { HttpMethod, HttpRequest, HttpRequestHeaders, HttpRequestParams, HttpRequestQuery } from '@azure/functions';
+import { AzureFunctionsRpcMessages as rpc } from '../../azure-functions-language-worker-protobuf/src/rpc';
+import { fromTypedData } from '../converters/RpcConverters';
+import { fromNullableMapping, fromRpcHttpBody } from '../converters/RpcHttpConverters';
 
-export class RequestProperties implements HttpRequest {
-    method: HttpMethod | null = null;
-    url = '';
-    originalUrl = '';
-    headers: { [key: string]: string } = {};
-    query: { [key: string]: string } = {};
-    params: { [key: string]: string } = {};
-    body?: any;
-    rawBody?: any;
-    [key: string]: any;
-}
+export class Request implements HttpRequest {
+    public method: HttpMethod | null;
+    public url: string;
+    public originalUrl: string;
+    public headers: HttpRequestHeaders;
+    public query: HttpRequestQuery;
+    public params: HttpRequestParams;
+    public body?: any;
+    public rawBody?: any;
 
-export class Request extends RequestProperties {
-    constructor(httpInput: RequestProperties) {
-        super();
-        Object.assign(this, httpInput);
+    public constructor(rpcHttp: rpc.IRpcHttp) {
+        this.method = <HttpMethod>rpcHttp.method;
+        this.url = <string>rpcHttp.url;
+        this.originalUrl = <string>rpcHttp.url;
+        this.headers = fromNullableMapping(rpcHttp.nullableHeaders, rpcHttp.headers);
+        this.query = fromNullableMapping(rpcHttp.nullableQuery, rpcHttp.query);
+        this.params = fromNullableMapping(rpcHttp.nullableParams, rpcHttp.params);
+        this.body = fromTypedData(<rpc.ITypedData>rpcHttp.body);
+        this.rawBody = fromRpcHttpBody(<rpc.ITypedData>rpcHttp.body);
     }
 
-    get(field: string): string | undefined {
+    public get(field: string): string | undefined {

```

Comment:
Just out of curiosity, what's the difference between specifying `public` and not having it at all? I see you added `public` to everything

Reply:
There's no difference. It's a stylistic thing that I'm used to doing. Eventually we should decide what we prefer and get the linter to automatically handle this for us. Until then, I think it's up to whatever the person wants